build/Microsoft.DotNet.Cli.BuildInfo.targets

```diff
@@ -0,0 +1,40 @@
+<Project ToolsVersion="15.0">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.tasks))\dir.tasks" />
+
+  <Target Name="WriteBuildInfoProps"
+          Condition=" !Exists('$(BuildInfoProps)') "
+          DependsOnTargets="BuildDotnetCliBuildFramework">
+    <GetCommitHash RepoRoot="$(RepoRoot)">
+      <Output TaskParameter="CommitHash" PropertyName="CommitHash" />
+    </GetCommitHash>
+
+    <GenerateBuildVersionInfo RepoRoot="$(RepoRoot)"
+                              VersionMajor="$(VersionMajor)"
+                              VersionMinor="$(VersionMinor)"
+                              VersionPatch="$(VersionPatch)"
+                              ReleaseSuffix="$(ReleaseSuffix)">
+      <Output TaskParameter="CommitCount" PropertyName="CommitCount" />
+      <Output TaskParameter="MsiVersion" PropertyName="MsiVersion" />
+      <Output TaskParameter="VersionBadgeMoniker" PropertyName="VersionBadgeMoniker" />
```

Comment:
I have no idea what that does. But I copied this from the prepare target so I assume we need it. Especially if I remove the call from prepare as @eerhardt suggested.